### PR TITLE
Panic when attempt to change size of a tuple

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
$subject.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java [hunk 1]
```diff
@@ -1072,15 +1072,19 @@ public IteratorValue getIterator() {
...
         checkFixedLength(length);
+        rangeCheck(length, size);
         fillerValueCheck(newLength, size);
         resizeInternalArray(newLength);
         fillValues(newLength);
         size = newLength;
     }
 
     private void checkFixedLength(long length) {
+        if (arrayType != null && arrayType.getTag() == TypeTags.TUPLE_TAG) {
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: Not all tuples should fail right? Shouldn't we be able to remove all from a tuple with only a rest field?

```
[int...] i = [2, 3];
i.removeAll(); // this is valid?
```

File: langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal [hunk 2]
```diff
@@ -249,3 +249,21 @@ function testRemoveAllFixedLengthArray() returns int[] {
     ar.removeAll();
     return ar;
 }
+
+function testTuppleResize() returns [int, string] {
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal]:
```suggestion
function testTupleResize() returns [int, string] {
```

Several other occurrences too.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java (hunk 1 above)
Comment (rdhananjaya) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: My understanding is that in above example [2,3] is a tuple value of type [int, int] and removing/setlength changes the type/shape of the value, IMO is illegal.

Comment (MaryamZi) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: The type is still `[int...]` right? The shape is `[int, int]`. Updates that change the shape can be done as long as they don't violate the inherent type (given the value is not immutable).

IMO, `[int...]` is the same as `int[]`.

Comment (rdhananjaya) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java]: Create a spec issue to clarify https://github.com/ballerina-platform/ballerina-spec/issues/309

File: bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties [hunk 3]
```diff
@@ -166,3 +166,4 @@ illegal.tuple.insertion = tuple of length {0} cannot be expanded into tuple of l
 unexpected.xml.type = lang.xml function ''{0}'' only accept xml type ''{1}''
 java.null.reference = handle refers to Java null
 illegal.array.size = cannot change length of fixed length array to ''{0}''
+cannot.change.tuple.size = cannot change tuple size
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/resources/MessagesBundle.properties]: Can we make this consistent with the array error message?

**Pull request merged**